pkg/gitparse/gitparse.go

```diff
@@ -309,8 +309,8 @@ func (c *Parser) FromReader(ctx context.Context, stdOut io.Reader, commitChan ch
 			if currentCommit == nil {
 				currentCommit = &Commit{}
 			}
-			if currentDiff != nil && currentDiff.Content.Len() > 0 {
-				currentCommit.Diffs = append(currentCommit.Diffs, *currentDiff)
+			if currentDiff.Content.Len() > 0 {
```

Comment:
why does making it an object allow us to avoid nil checking? is it just going to default to an empty struct?

Reply:
Yep. When its initialized it is an empty struct instead of a pointer to `nil`.

Reply:
In this case @bill-rich would it be a pointer to nil or a nil pointer? I think the latter?
This is my understanding:
```
var i *int = nil // i is a nil pointer
var p **int = &i // p is a pointer to a nil pointer || closest thing to pointer to nil lol
```

Reply:
You're right. I worded that poorly. Should have been "pointing to `nil`".